conseil-lorre/src/main/scala/tech/cryptonomic/conseil/indexer/tezos/TezosIndexer.scala

```diff
@@ -518,446 +174,151 @@ class TezosIndexer(
-      ): Future[List[(BlockTagged[Map[AccountId, Account]], List[Tables.AccountsRow])]] =
-        Future.traverse(taggedAccounts) { blockTaggedAccounts =>
-          if (blockTaggedAccounts.blockLevel % lorreConf.blockRightsFetching.cycleSize == 1) {
-            nodeOperator.fetchActiveBakers(taggedAccounts.map(x => (x.blockLevel, x.blockHash))).flatMap {
-              activeBakers =>
-                val activeBakersIds = activeBakers.toMap.apply(blockTaggedAccounts.blockHash)
-                db.run {
-                  TezosDb
-                    .getInactiveBakersFromAccounts(activeBakersIds)
-                    .map(blockTaggedAccounts -> _)
+  /** * Creates the Indexer which is dedicated for Tezos BlockChain */
+  def fromConfig(
+      lorreConf: LorreConfiguration,
+      conf: TezosConfiguration,
+      callsConf: NetworkCallsConfiguration,
+      streamingClientConf: HttpStreamingConfiguration,
+      batchingConf: BatchFetchConfiguration
+  ): LorreIndexer = {
+    val selectedNetwork = conf.network
+
+    implicit val system: ActorSystem = ActorSystem("lorre-tezos-indexer")
+    implicit val materializer: ActorMaterializer = ActorMaterializer()
+    implicit val dispatcher: ExecutionContext = system.dispatcher
+
+    val ignoreProcessFailuresOrigin: Option[String] = sys.env.get(LORRE_FAILURE_IGNORE_VAR)
+    val ignoreProcessFailures: Boolean =
+      ignoreProcessFailuresOrigin.exists(ignore => ignore == "true" || ignore == "yes")
+
+    /* Here we collect all internal service operations and resources, needed to run the indexer */
+    val db = DatabaseUtil.lorreDb
+    val indexedData = new TezosIndexedDataOperations
+
+    /* collects data from the remote tezos node */
+    val nodeOperator = new TezosNodeOperator(
+      new TezosNodeInterface(conf, callsConf, streamingClientConf),
+      selectedNetwork,
+      batchingConf,
+      indexedData
+    )
+
+    /* provides operations to handle rights to bake and endorse blocks */
+    val rightsProcessing = new BakingAndEndorsingRightsProcessing(
+      db,
+      lorreConf.blockRightsFetching,
+      nodeOperator,
+      indexedData
+    )
+
+    /* handles standard accounts data */
+    val accountsProcessing = new AccountsProcessing(nodeOperator, db, batchingConf, lorreConf.blockRightsFetching)
+
+    /* handles occasional global accounts refreshes due to very specific events */
+    val accountsEventsProcessing = new AccountsEventsProcessing(db, indexedData)
+
+    /* handles bakers data */
+    val bakersProcessing = new BakersProcessing(nodeOperator, db, batchingConf, lorreConf.blockRightsFetching)
+
+    /* Reads csv resources to initialize db tables and smart contracts objects */
+    def initAnyCsvConfig(): TokenContracts = {
+      import kantan.csv.generic._
+      import tech.cryptonomic.conseil.common.util.ConfigUtil.Csv._
+
+      /* Inits tables with values from CSV files */
+      TezosDb.initTableFromCsv(db, Tables.KnownAddresses, selectedNetwork)
```

Comment:
Our of curiosity, where is the implicit `HeaderDecoder` here coming from? Maybe it's possible to leave a comment?

Reply:
will do